pyhealth/datasets/mimicextract.py

```diff
@@ -0,0 +1,475 @@
+        return patients
+
+    def parse_interventions(self, patients: Dict[str, Patient]) -> Dict[str, Patient]:
+        table = 'interventions'
+        linksto = table.lower() # we might put these in CHARTEVENTS also?
+
+        # read table
+        df = pd.read_hdf(self._ahd_filename, 'interventions')
+        # drop records of the other patients
+        df = df.loc[(list(patients.keys()),slice(None),slice(None)),:]
+
+        # parallel unit for interventions (per patient)
+        captured_v_id_column = self._v_id_column
+        captured_vocab_map = self._vocab_map[linksto] 
+        def interv_unit(p_id, p_info):
+            events = []
+            for v_id, v_info in p_info.groupby(captured_v_id_column):
+                for e_id, e_info in v_info.iterrows():
+                    event = Event(
+                        code=captured_vocab_map[e_info['variable']],
+                        table=table,
+                        vocabulary="MIMIC3_ITEMID",
+                        visit_id=v_id,
+                        patient_id=p_id,
+                        timestamp=pd.Timestamp.to_pydatetime(e_info['timestamp']),
+                        hours_in=int(e_info['hours_in']),
+                        intervention=e_info['variable']
+                    )
+                    events.append(event)
+            return events
+
+        ahd_index = ["subject_id","hadm_id","icustay_id","hours_in"]
+
+        #if 'LEVEL1' in df.columns.names:
+        #    df.columns = df.columns.get_level_values(2)
+        #else:
+        #    df.columns = df.columns.get_level_values(0)
+
+        # reconstruct nominal timestamps for hours_in values...
+        df_p = pd.read_hdf(self._ahd_filename, 'patients')
+        df_p = df_p.loc[(list(patients.keys()),slice(None),slice(None)),:][['intime']]
+        df = df.merge(df_p, left_on=['subject_id','hadm_id','icustay_id'], right_index=True, how="left")
+        df['timestamp'] = df['intime'].dt.ceil('H')+pd.to_timedelta(df.index.get_level_values(3), unit="H")
+
+        df = df.drop(columns=[col for col in df.columns if col not in self._vocab_map[linksto] and col not in ['timestamp']])
+        df = df.reset_index()
+        df = df.melt(id_vars=ahd_index+['timestamp'])
+        df = df[df['value'] > 0]
+        df = df.sort_values(ahd_index)
+        group_df = df.groupby("subject_id")
+
+        # parallel apply
+        group_df = group_df.parallel_apply(
+            lambda x: interv_unit(x.subject_id.unique()[0], x)
+        )
+
+        # summarize the results
+        patients = self._add_events_to_patient_dict(patients, group_df)
+        return patients
+
+
+if __name__ == "__main__":
+    dataset = MIMICExtractDataset(
+        root="../data/baseline5000/grouping",
+        tables=[
+            "C",
+            "vitals_labs",
+            "interventions"
+        ],
+        #code_mapping={"NDC": "ATC"},
+        dev=True,
+        refresh_cache=True,
+    )
```

Comment:
Could you give me a link to the MIMIC-Extract dataset? Also, could you send a pr (not draft), so that I can merge your code to pyhealth/mimic-extract for local testing. If everything is fixed, I will merge into master.

Reply:
I've uploaded two datasets (different grouping options) here:
https://drive.google.com/drive/folders/1RaioXfcHYKUwyL__LdUcbVnmLXrgDUmp?usp=sharing

I've also included a copy of the `itemid_to_variable_map.csv` file from MIMIC-Extract in there for convenience.